spec-0007/index.md

```diff
@@ -161,62 +186,137 @@ def check_random_state(seed=_NoValue, rng=_NoValue):
         return seed
 
     raise ValueError(f"'{seed}' cannot be used to seed a numpy.random.RandomState"
-                     " instance")
-
-
-def _prepare_rng(old_name, dep_version=None):
+                     " instance.")
+
+
+def _prepare_rng(old_name, position_num=None, dep_version=None):
+    """ Example decorator to deprecate old PRNG usage
+
+    Suppose the decorator is applied to a function that used to accept parameter
+    `old_name='random_state'` either by keyword or as a positional argument at
+    `position_name=1`. At the time of application, the name of the argument in the
+    function signature is manually changed to the new name, `rng`.
+    - If the function is called with both `random_state` and `rng`, the decorator
+      raises an error.
+    - If `random_state` is provided as a keyword argument, the decorator validates
+      the argument according to the old rules* and passes it to the function as `rng`,
```

Comment:
Oops, I meant to specify what `_check_random_state` assumes the "old rules" to be:
- `None` becomes the global `RandomState`.
- integers seed a new `RandomState`
- `RandomState` or `Generator` instances are passed through unchanged
- Everything else is rejected.

If this is not what the wrapped function used to do, the decorator/`_check_random_state` cannot be applied as-is, so I think it's important to document this.

Where should this list go?

Reply:
No longer needed.